test/default-route.test.js

```diff
@@ -4,6 +4,26 @@ const t = require('tap')
 const test = t.test
 const Fastify = require('..')
 
+test('setDefaultRoute should emit a deprecation warning', t => {
+  t.plan(2)
+
+  const fastify = Fastify()
+  const defaultRoute = (req, res) => {
+    res.end('hello from defaultRoute')
+  }
+
+  process.on('warning', onWarning)
+  function onWarning (warning) {
+    t.equal(warning.name, 'FastifyDeprecation')
+    t.equal(warning.code, 'FSTDEP014')
+  }
+
+  t.teardown(() => process.removeListener('warning', onWarning))
+
+  fastify.setDefaultRoute(defaultRoute)
+  fastify.getDefaultRoute()
```

Comment:
Due to the nature of `process-warning`. It won't emit a second warning for the same code, so I can't test `getDefaultRoute` after `setDefaultRoute` (or vice-versa). A possible solution would be spawning a new process or deleting `require.cache[path.resolve('../fastify.js']` and importing again. 

However, I don't think that's necessary,  so I've just included the second call to avoid a future mistake.

Reply:
See https://github.com/ldapjs/messages/blob/6a680528f850d3a3b9044e77c89f4845cfdec9b3/lib/ldap-message.test.js#L28-L51

Reply:
Oh good one. Thanks.